clients/src/test/java/org/apache/kafka/clients/consumer/internals/PrototypeAsyncConsumerTest.java

```diff
@@ -48,72 +40,64 @@
-import static org.apache.kafka.clients.consumer.ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG;
-import static org.apache.kafka.clients.consumer.ConsumerConfig.DEFAULT_API_TIMEOUT_MS_CONFIG;
-import static org.apache.kafka.clients.consumer.ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG;
-import static org.apache.kafka.clients.consumer.ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG;
 import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
 import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
 import static org.junit.jupiter.api.Assertions.assertThrows;
 import static org.junit.jupiter.api.Assertions.assertTrue;
 import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
 import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.doThrow;
 import static org.mockito.Mockito.mock;
 import static org.mockito.Mockito.mockConstruction;
 import static org.mockito.Mockito.never;
-import static org.mockito.Mockito.spy;
 import static org.mockito.Mockito.verify;
 import static org.mockito.Mockito.when;
 
 public class PrototypeAsyncConsumerTest {
 
-    private Consumer<?, ?> consumer;
-    private final Map<String, Object> consumerProps = new HashMap<>();
+    private static final Optional<String> DEFAULT_GROUP_ID = Optional.of("group.id");
 
-    private final Time time = new MockTime();
-    private LogContext logContext;
-    private SubscriptionState subscriptions;
+    private ConsumerTestBuilder.PrototypeAsyncConsumerTestBuilder consumerBuilder;
     private EventHandler eventHandler;
-    private Metrics metrics;
-    private String groupId = "group.id";
-    private ConsumerConfig config;
+    private PrototypeAsyncConsumer<String, String> consumer;
 
     @BeforeEach
     public void setup() {
-        injectConsumerConfigs();
-        this.config = new ConsumerConfig(consumerProps);
-        this.logContext = new LogContext();
-        this.subscriptions = mock(SubscriptionState.class);
-        this.eventHandler = mock(DefaultEventHandler.class);
-        this.metrics = new Metrics(time);
+        setup(DEFAULT_GROUP_ID);
     }
 
     @AfterEach
     public void cleanup() {
-        if (consumer != null) {
-            consumer.close(Duration.ZERO);
-        }
+        if (consumer != null)
+            consumer.close();
+    }
+
+    private void setup(Optional<String> groupIdOpt) {
+        consumerBuilder = new ConsumerTestBuilder.PrototypeAsyncConsumerTestBuilder(groupIdOpt);
+        eventHandler = consumerBuilder.eventHandler;
+        consumer = consumerBuilder.consumer;
     }
 
     @Test
     public void testSuccessfulStartupShutdown() {
-        consumer = newConsumer(time, new StringDeserializer(), new StringDeserializer());
-        assertDoesNotThrow(() -> consumer.close());
+        assertDoesNotThrow(() -> consumerBuilder.consumer.close());
```

Comment:
nit nit: all tests use the local consumer instance instead of the consumerBuilder.consumer, so probably better to keep this one alike, for consistency

Reply:
I thought I caught all of these cases, but alas...